lib/compute/jenkins-main-node.ts

```diff
@@ -360,72 +345,39 @@ export class JenkinsMainNode {
             },
           },
         },
-        logs: {
-          logs_collected: {
-            files: {
-              collect_list: [
-                {
-                  file_path: '/var/log/jenkins/jenkins.log',
-                  log_group_name: 'JenkinsMainNode/var/log/jenkins/jenkins.log',
-                  auto_removal: true,
-                  log_stream_name: 'jenkins.log',
-                  //  2021-07-20 16:15:55.319+0000 [id=868]   INFO    jenkins.InitReactorRunner$1#onAttained: Completed initialization
-                  timestamp_format: '%Y-%m-%d %H:%M:%S.%f%z',
-                },
-              ],
-            },
-          },
-          force_flush_interval: 15,
-        },
       }),
-
       InitCommand.shellCommand('/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a stop'),
       // eslint-disable-next-line max-len
       InitCommand.shellCommand('/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -c file:/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json -s'),
 
-      // start jenkins service to generate all the default files and folders of jenkins
-      // Does not use InitService.enable() because it initialises the service after the instance is launched
-      InitCommand.shellCommand('systemctl start jenkins'),
+      InitCommand.shellCommand(dataRetentionProps.dataRetention
+        ? `mkdir /var/lib/jenkins && mount -t efs ${efsId} /var/lib/jenkins`
+        : 'echo Data rentention is disabled, not mounting efs'),
+
+      InitFile.fromFileInline('/docker-compose.yml', join(__dirname, '../../resources/docker-compose.yml')),
+      InitCommand.shellCommand('systemctl start docker && docker-compose up -d'),
 
       // Commands are fired one after the other but it does not wait for the command to complete.
       // Therefore, sleep 60 seconds to wait for jenkins to start
-      // This allows jenkins to generate the secrets files used for auth in jenkins-cli APIs
       InitCommand.shellCommand('sleep 60'),
```

Comment:
given we have removed the commands, do we need the sleep 60?

Reply:
Yes to let jenkins come up via container